tests/e2e/rayjob/lifecycled_cluster_oauth_test.py

```diff
@@ -58,113 +51,95 @@ def run_rayjob_with_lifecycled_cluster_oauth(self):
+            assert job_cr["spec"]["suspend"] is False, "Job suspend not set to false"
+
+            assert self.job_api.wait_until_job_finished(
+                name=rayjob.name, k8s_namespace=rayjob.namespace, timeout=300
+            ), "Job did not complete"
+
+        finally:
+            # 4. Delete the job and cleanup
+            assert rayjob.delete_job()
+            self.verify_cluster_cleanup(rayjob)
+
+    def _wait_for_job_status(
+        self,
+        rayjob: RayJob,
+        expected_status: str,
+        timeout: int = 30,
+    ) -> bool:
+        """Wait for a job to reach a specific deployment status."""
         elapsed_time = 0
-        check_interval = 10  # Check every 10 seconds
+        check_interval = 2
 
         while elapsed_time < timeout:
-            status, ready = rayjob.status(print_to_console=True)
-
-            # Check if job has completed (either successfully or failed)
-            if status == CodeflareRayJobStatus.COMPLETE:
-                print(f"RayJob '{rayjob.name}' completed successfully!")
-                return
-            elif status == CodeflareRayJobStatus.FAILED:
-                raise AssertionError(f"RayJob '{rayjob.name}' failed!")
-            elif status == CodeflareRayJobStatus.RUNNING:
-                print(f"RayJob '{rayjob.name}' is still running...")
-            elif status == CodeflareRayJobStatus.UNKNOWN:
-                print(f"RayJob '{rayjob.name}' status is unknown")
-
-            # Wait before next check
+            status = self.job_api.get_job_status(
+                name=rayjob.name, k8s_namespace=rayjob.namespace
+            )
+            if status and status.get("jobDeploymentStatus") == expected_status:
+                return True
+
             sleep(check_interval)
             elapsed_time += check_interval
 
-        # If we reach here, the job has timed out
-        final_status, _ = rayjob.status(print_to_console=True)
-        raise TimeoutError(
-            f"RayJob '{rayjob.name}' did not complete within {timeout} seconds. "
-            f"Final status: {final_status}"
-        )
-
-    def verify_cluster_cleanup(self, cluster_name: str, timeout: int = 60):
-        """
-        Verify that the cluster created by the RayJob has been cleaned up.
-        Args:
-            cluster_name: The name of the cluster to check for cleanup
-            timeout: Maximum time to wait for cleanup in seconds (default: 1 minute)
-        """
-        from kubernetes import client
-        import kubernetes.client.rest
+        return False
 
+    def verify_cluster_cleanup(self, rayjob: RayJob, timeout: int = 60):
+        """Verify that the cluster created by the RayJob has been cleaned up."""
         elapsed_time = 0
-        check_interval = 5  # Check every 5 seconds
+        check_interval = 5
+        cluster_api = RayClusterApi()
 
         while elapsed_time < timeout:
             try:
-                # Try to get the RayCluster resource
```

Comment:
@pawelpaszki I moved to using the ClusterAPi Python client where possible since that's wrapping the k8s api. just for future reference with these tests if we are interacting with job or cluster CRs we should just use the upstream clients